src/mono/mono/utils/mono-threads-wasm.c

```diff
@@ -75,7 +72,7 @@ wasm_get_stack_size (void)
 static int
 wasm_get_stack_base (void)
 {
-	return get_wasm_data_end();
+	return get_wasm_stack_high();
```

Comment:
should we do something like:
```c
#ifndef DISABLE_THREADS
#error "get_wasm_stack_high" isn't multi-thread aware
#endif
```
?

Reply:
I'll let Pavel decide :)

Reply:
there is some support for this in later versions of LLVM. I don't remember the details right now. 
I will start looking into WASI soon.

Reply:
> And yes, I think this is not just for the UI/main thread.

You're right. I forgot how wasm globals work before the "shared everything "proposal". they're per-instance.  so every thread has a separate global which points to its portion of linear memory. so it's all good.